pkg/apis/pipeline/v1beta1/resultref.go

```diff
@@ -122,12 +129,16 @@ func stripVarSubExpression(expression string) string {
 	return strings.TrimSuffix(strings.TrimPrefix(expression, "$("), ")")
 }
 
-func parseExpression(substitutionExpression string) (string, string, error) {
+func parseExpression(substitutionExpression string) (string, string, string, error) {
 	subExpressions := strings.Split(substitutionExpression, ".")
-	if len(subExpressions) != 4 || subExpressions[0] != ResultTaskPart || subExpressions[2] != ResultResultPart {
-		return "", "", fmt.Errorf("Must be of the form %q", resultExpressionFormat)
+	if len(subExpressions) < 4 || len(subExpressions) > 5 || subExpressions[0] != ResultTaskPart || subExpressions[2] != ResultResultPart {
```

Comment:
this expression is difficult to read-- any chance you can add comments/helpful variable names/split into multiple lines or any of the above?

Reply:
> this expression is difficult to read-- any chance you can add comments/helpful variable names/split into multiple lines or any of the above?

Great point. Just separated them and added comments!